roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -1911,15 +1911,55 @@ RooArgSet* RooProdPdf::findPdfNSet(RooAbsPdf const& pdf) const
 }
 
 
+
+/// Add some full PDFs to the factors of this RooProdPdf.
+void RooProdPdf::addPdfs(RooArgSet const& pdfs)
+{
+   size_t numExtended = (_extendedIndex==-1) ? 0 : 1;
+
+   for(auto arg : pdfs) {
+      RooAbsPdf* pdf = dynamic_cast<RooAbsPdf*>(arg);
```

Comment:
The `RooProdPdf` constructor prints a warning here in case it's not a pdf. Why don't you do the same here? Probably there are usecases where some elements in the argument set, and you just want to ignore them?

A compromise should be a info printout with `coutI`. In any case, it should be consistent with the constructor (see my other comment about code duplication).

Reply:
thanks for suggestion to use this new method in the constructor so that we get test coverage. I've implemented the messaging from that migration now as well